### PR TITLE
Reworked Vitess error codes

### PR DESCRIPTION
**Pull request by aaijazi**
@alainjobart

### CODE REVIEW

File: proto/vtrpc.proto [hunk 1]
```diff
@@ -59,6 +106,6 @@ enum ErrorCode {
 // We use this so the clients don't have to parse the error messages,
 // but instead can depend on the value of the code.
 message RPCError {
```
Comment (alainjobart) [inline proto/vtrpc.proto]: What other name will you use for the remote errors we send over rpc?
(just wondering if we need to rename this one too...)

File: proto/vtrpc.proto [hunk 2]
```diff
@@ -33,8 +33,55 @@ message CallerID {
   string subcomponent = 3;
 }
 
-// ErrorCode is the enum values for Errors
+// ErrorCode is the enum values for Errors. Internally, errors should
+// be created with one of these codes. These will then be translated over the wire
+// by various RPC frameworks.
 enum ErrorCode {
+  SUCCESS = 0;
```
Comment (alainjobart) [inline proto/vtrpc.proto]: I would use the same doc style as a go enum, with one comment per enum too:
// SUCCESS means ...

File: proto/vtrpc.proto [hunk 3]
```diff
@@ -33,8 +33,55 @@ message CallerID {
...
+  // Context was cancelled (and noticed in the app layer, as opposed to the RPC layer)
+  CANCELLED = 1;
+  // MySQL error code that we don't explicitly handle
+  MYSQL_ERROR = 2;
```
Comment (alainjobart) [inline proto/vtrpc.proto]: See comment on design doc, we need one value per canonical error code in RPC framework, not more...

File: proto/vtrpc.proto (hunk 1 above)
Comment (aaijazi) [inline proto/vtrpc.proto]: I don't know yet. I planned on punting on this for now, and thinking about it more carefully when devising a roll out plan, so I don't accidentally do unhappy things to backwards compatibility.

File: proto/vtrpc.proto (hunk 3 above)
Comment (aaijazi) [inline proto/vtrpc.proto]: (discussion continued in the doc)

File: proto/vtrpc.proto (hunk 2 above)
Comment (aaijazi) [inline proto/vtrpc.proto]: Done

**Comment by aaijazi** [comment]
PTAL

**Comment by alainjobart** [comment]
LGTM

**Pull request merged**